e2e/image/build_test.go

```diff
@@ -103,6 +108,33 @@ func TestTrustedBuildUntrustedImage(t *testing.T) {
 	})
 }
 
+func TestBuildIidFileSquash(t *testing.T) {
+	environment.SkipIfNotExperimentalDaemon(t)
+	dir := fs.NewDir(t, "test-iidfile-squash")
+	defer dir.Remove()
+	iidfile := filepath.Join(dir.Path(), "idsquash")
+	buildDir := fs.NewDir(t, "test-iidfile-squash-build",
+		fs.WithFile("Dockerfile", fmt.Sprintf(`
+	FROM %s
+	ENV FOO FOO
+	ENV BAR BAR
+	RUN touch /foop`, fixtures.AlpineImage)),
```

Comment:
Looks like this Dockerfile will only add a single layer; should we have something that actually has multiple layers before squashing?

(If there's some specific case we're testing for, we may need a comment to describe that case)

Reply:
Oh good point. I took the test case from `moby/moby` but yeah we could add one more `RUN` to it :wink:

Reply:
The original issue was https://github.com/moby/moby/issues/33822, which lead to https://github.com/moby/moby/pull/33824.

Rereading that I think the number of layers was not the issue, just the wrong thing was being returned.

Reply:
Ah, thanks!

@vdemeester @ijc do think it would be good to add that as a comment/godoc for the test, so that the original intent doesn't get lost?